.github/workflows/gtn-news.yml

```diff
@@ -0,0 +1,162 @@
+
+          import feedparser
+
+          feedparser.sanitizer._HTMLSanitizer.acceptable_elements.remove("a")
+          feedparser.sanitizer._HTMLSanitizer.acceptable_elements.remove("img")
+
+          feed = feedparser.parse("https://training.galaxyproject.org/training-material/feed.xml")
+
+          posts = {}
+          oldest = datetime.now().astimezone()
+          for entry in feed.entries:
+              feed_title = f"[GTN news] {entry['title']}"
+              feed_link = entry["link"]
+              feed_blurb = entry["summary"]
+              feed_published = entry["published"].split("T")[0]
+              feed_name = feed_link.split("/")[-1].replace(".html", "")
+              posts[f"{feed_published}-{feed_name}"] = {
+                  "title": feed_title,
+                  "link": feed_link,
+                  "blurb": feed_blurb,
+                  "published": feed_published,
+                  "name": feed_name,
+              }
+              feed_published_dt = datetime.fromisoformat(entry["published"])
+              oldest = feed_published_dt if feed_published_dt < oldest else oldest
+          oldest = (oldest - timedelta(days=1)).astimezone(timezone.utc).date()
+
+          with open(os.environ["GITHUB_OUTPUT"], 'w') as file:
+              print("posts=%s" % json.dumps(posts), file=file)
+              print("date=%s" % oldest, file=file)
+
+      # The next two steps prevent the `pull_requests` job from running when the target pull request may
+      # already exist. They are not strictly necessary because the GitHub Action
+      # `peter-evans/create-pull-request@v5` does not create duplicate PRs by default
+      # (see https://github.com/peter-evans/create-pull-request#action-behaviour)
+      - name: Get files touched by existing GitHub PRs that are newer than the oldest article in the feed.
+        id: files
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          files="$(gh pr list --repo https://github.com/${{ github.repository }} -s all -S "created:>=${{ steps.parse.outputs.date }}" --json files)"
+          echo files="$files" > $GITHUB_OUTPUT
+      - name: Filter out GTN news for which a PR may already exist.
+        id: filter
+        shell: python -u {0}
+        run: |
+          import json
+          import os
+
+          files=json.loads("""
+              ${{ steps.files.outputs.files }}
+          """)
+          files={file["path"] for item in files for file in item["files"]}
+
+          posts=json.loads("""
+              ${{ steps.parse.outputs.posts }}
+          """)
+
+          existing = {
+              name
+              for name in posts
+              if any(name in file for file in files)
+          }
+          for name in existing:
+              del posts[name]
+
+          with open(os.environ["GITHUB_OUTPUT"], 'w') as file:
+              print("posts=%s" % json.dumps(posts), file=file)
+
+      - name: Prepare matrix for `pull_requests` job.
```

Comment:
this seems complicated, why not have one PR that adds any (multiple) missing posts? it would be less work to manage than multiple PRs to approve and merge

Reply:
PRs don't necessarily mean a lot of extra work, it's just more transparent vs committing to master.  Other than that these should be PRs, I don't have a preference over single or batched -- my guess is that only matters for the initial pass here and that they'll be largely single posts past that.

Reply:
I guess at least initially we should go for PRs for transparency. We can later auto-merge those after a given time or commit directly - which would save some CPU cycles and energy. But lets at first see if everything works.

Are we fine with merging as it is or doing the bulk-merge?
@dannon your call :)

Reply:
yeah PRs is good, just a single batch one would simplify this GHA *significantly* e.g. the [gtn's updater](https://github.com/galaxyproject/training-material/blob/2dd1a79f2e9578eee751c4a57f4616e772a0fbe6/.github/workflows/shortlinks.yaml#L43)